### PR TITLE
[12.0][ADD] pos_ticket_partner_firstname

### PR DESCRIPTION
**Pull request by ivantodorovich**
On the POS ticket, prints only the salesman's firstname.

receipt

### CODE REVIEW

File: pos_ticket_partner_firstname/static/src/xml/pos.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<templates id="template" xml:space="preserve">
+
+    <t t-extend="PosTicket">
```
Comment (legalsylvain) [inline pos_ticket_partner_firstname/static/src/xml/pos.xml]: until Odoo 13 (I think) there are two ticket format. ``PosTicket`` and the ``XmlReceipt``. 
could you handle both cases in your module? 

https://github.com/odoo/odoo/blob/12.0/addons/point_of_sale/static/src/xml/pos.xml#L736

if it's not in the scope of your developpement, please add a know issues in the roadmap file to mention that this module does'nt work in a non proxy mode.

thanks !

**Review by legalsylvain — COMMENTED** [review]
Hi @ivantodorovich thanks for your contribution.

a remarks, regarding the ticket. otherwise LGTM.
I was asking to myself if if was not better to add this option in a existing glue module https://github.com/OCA/pos/tree/12.0/pos_partner_firstname introduced by https://github.com/OCA/pos/pull/430


what do you think @eLBati ?

kind regards.

**Comment by eLBati** [comment]
@legalsylvain `pos_partner_firstname` handles first name of customers, the module of this PR handles first name of salesman. I think they have different scopes.

I would rather suggest to rename this module to `pos_ticket_salesman_firstname`

**Comment by legalsylvain** [comment]
OK @eLBati. Thanks for your point of view !

**Comment by ivantodorovich** [comment]
Thanks @eLBati !
I agree, `pos_ticket_salesman_firstname` is more clear :+1: 
I've renamed the module

File: pos_ticket_partner_firstname/static/src/xml/pos.xml (hunk 1 above)
Comment (ivantodorovich) [reply pos_ticket_partner_firstname/static/src/xml/pos.xml]: It's already handled, by overloading `export_for_printing` on the js file :)
`XmlReceipt` gets the cashier.name from that function

Comment (legalsylvain) [reply pos_ticket_partner_firstname/static/src/xml/pos.xml]: thanks !

**Review by legalsylvain — APPROVED** [review]
@elbati, is it now OK for you ? 

@ivantodorovich : thanks. LGTM. (code review / no Test)

**Comment by eLBati** [comment]
@ivantodorovich could you squash the 2 commits into 1?

**Review by eLBati — CHANGES_REQUESTED** [review]
Testing on runbot with `Print only the salesman's first name` set, the name of the user is not displayed:

image

@ivantodorovich do you know why?

**Comment by ivantodorovich** [comment]
> Testing on runbot with `Print only the salesman's first name` set, the name of the user is not displayed:
> 
> image
> 
> @ivantodorovich do you know why?

My bad. Forgot to change the assets view after renaming the module.
It should be okay now

**Comment by ivantodorovich** [comment]
> @ivantodorovich could you squash the 2 commits into 1?

done :+1:

**Review by eLBati — APPROVED** [review]
Thanks

**Pull request merged**